src/docs/arch-hs.dox

```diff
@@ -1,8 +1,272 @@
+  </tr>
+  <tr style="height: 32px;">
+    <td style="width: 72.5156px; height: 32px; text-align: center;">1000</td>
+    <td style="width: 89.875px; height: 32px; text-align: center;">"AAA"</td>
+    <td style="width: 97.0312px; height: 32px; text-align: center;">90</td>
+    <td style="width: 93.9688px; height: 32px; text-align: center;">0</td>
+    <td style="width: 92.4375px; height: 32px; text-align: center;">100</td>
+    <td style="width: 105.719px; height: 32px; text-align: center;">90</td>
+    <td style="width: 110.312px; height: 32px; text-align: center;">STANDARD</td>
+    <td style="width: 148.141px; height: 32px; text-align: center;">Value from U3</td>
+  </tr>
+  <tr style="height: 32px;">
+    <td style="width: 72.5156px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 89.875px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 97.0312px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 93.9688px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 92.4375px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 105.719px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 110.312px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 148.141px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+  </tr>
+</table>
+
+For the history store table entry corresponding to the update \c U1, the start timestamp in the key
+is the start timestamp for the update \c U1 and the stop timestamp in the value is the start
+timestamp for the update \c U2.
+
+When a modified history store table page is processed by reconciliation, certain records on the page
+are removed when creating a new page image if the tombstones on those records have become globally
+visible. This ensures that WiredTiger is only keeping the relevant historical data required to
+serve the oldest reader in the system or as dictated by the oldest timestamp set by the application.
+Once all records on a page are obsolete, the page itself can be removed to reduce the size of
+the history store table (see @ref arch-transaction for more details about oldest timestamp).
+
+@section arch_hs_read Searching for older versions of a key in History Store
+When looking for an update visible to the current transaction, WiredTiger first searches the update
+chain for any visible updates. If there is no visible update in the chain, WiredTiger then
+inspects the on-disk version of the key. If that version is not visible to the transaction,
+WiredTiger searches the history store table for a visible update for the key. When
+searching for update that satisfies the read timestamp constraint, WiredTiger starts with the newest
```

Comment:
There might be a word missing here:
`When searching for *an* update`

Reply:
Fixed.